src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -28,24 +28,19 @@
 #include "logging/logHandle.hpp"
 #include "runtime/atomic.hpp"
 
-Semaphore AsyncLogWriter::_sem(0);
-Semaphore AsyncLogWriter::_io_sem(1);
-
 class AsyncLogLocker : public StackObj {
- private:
-  static Semaphore _lock;
+  Semaphore* _sem;
+
  public:
-  AsyncLogLocker() {
-    _lock.wait();
+  AsyncLogLocker(Semaphore* sem): _sem(sem) {
+    _sem->wait();
```

Comment:
This class is not general purpose. It does not need to accept an arbitrary semaphore. There is only one semaphore this class works with and that is AsyncLogWriter::_instance->_lock, so it can just use it directly.

Reply:
Update.  declare AsyncLogLocker as an inner class so we don't need declare it as a friend.